built_value_generator/test/serializer_generator_test.dart

```diff
@@ -226,6 +226,27 @@ abstract class _Value implements Built<_Value, _ValueBuilder> {
 }
 '''), contains(r'1. Cannot generate serializers for private class _Value'));
     });
+
+    test('deserialize is null safe', () async {
```

Comment:
Thanks for the PR :)

Please add this as an end to end test rather than a unit test; the setup is that correct usage is tested as end to end tests, so we test the actual effect of the feature, and wrong usage is tested as unit tests, so we test the error output.

To add as an end to end test:

Run tool/use_local_deps so the end to end test uses the updated version of the generator. In the end_to_end_test folder you can now run the generation to iterate.

Add the lint to analysis_options.yaml for end_to_end_test. The lints are checked on presubmit, so this is now a test for the feature. Probably you don't need to add any new actual end to end cases to cover the feature.

Once it looks like it's working, try tool/presubmit. That regenerates everything, checks that it passes analysis (including lints), and runs all the tests. It's a bit slow though :)

Reply:
I had to change a couple' small things and got the tool/presubmit to pass. That's a pretty neat test setup :)

Should I update the PR with newly generated code (after tool/presubmit) and the analysis_options change too, or just the little change I had to make to make tool/presubmit pass?

Reply:
Please include everything in the pr :) thanks.

Reply:
Updated the PR. Thanks!

Reply:
Also, the presubmit tool seems to be using dartfmt, pub, and dartanalyzer which seem to be replaced with 'dart format', 'dart pub', and 'dart analyze' for the outside world. I think the recent push also contains these changes. Let me know if that's not ideal and I'll change the presuibmit tool script back.

Reply:
Works for me, thanks :)

Change looks good, please also add to the changelog, something like:

# 8.2.0 (unreleased)

- Make generator output additional explicit null checks so the generated code complies with the `cast_nullable_to_non_nullable` lint.

Reply:
Copy pasta and.... updated! Thanks! :)

Reply:
Ah, sorry, one more, please remove the unit test :) ... the e2e test has this covered now, and unit tests on generated code are quite painful to maintain :) thanks!

Reply:
Done, and no worries, thanks for being patient and pointing out all these little changes. :)

Reply:
LG, thanks :)